api.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+	"log"
+	"net/http"
+)
+
+/*
+ * Returns a HelloWorld json message
+ */
+func helloHandler(c *gin.Context) {
+	c.JSON(http.StatusOK, gin.H{"message": "Hello World!"})
+}
+
+/*
+ * Returns a greeting json message
+ */
+func greetHandler(c *gin.Context) {
+	name := c.Param("name")
+	c.JSON(http.StatusOK, gin.H{"greet-message": "Hello " + name + "!"})
+}
+
+/*
+ * Returns a json array of all the products
+ */
+func GetAllPatients(c *gin.Context) {
+
+	rows, err := db.Query("SELECT * FROM patient")
```

Comment:
You should be using `sqlx` for this. The `db` should be of type `*sqlx.DB`, so you can do `db.Select()`, which is easier to read. https://jmoiron.github.io/sqlx/

Reply:
Thanks for it, it reduced my coding errorts and lines to almost half :)